File: spoj/CANTON.cpp
#include<cstdio>
#include<math.h>
#include<iostream>
using namespace std;
int main()
{
    int Test_case;
    scanf("%d",&Test_case);
    while(Test_case--)
    {
        int i,Num,sums=0;
        scanf("%d",&Num);
        for(i=1;;i++)
        {
            sums+=i;
            if(sums>=Num)
                break;
        }
        int temp=Num-(sums-i);
        int total=i+1;
        if(i%2==0)
            printf("TERM %d IS %d/%d\n",Num,temp,total-temp);
        else
            printf("TERM %d IS %d/%d\n",Num,total-temp,temp);
    }
    return 0;
}

File: spoj/CANTON.py
T = int(raw_input())
import math.sqrt as sqp
while T>0:
	N = int(raw_input())
	x = sqp(8*N+1)/2
	t = (x*x+x)/2
	if x%2==0:
		k=t - N+1
		r = t-x+1
		print "TERM", N, "IS", k, "/", r
	else:
		r = 
